climada/util/coordinates.py

```diff
@@ -1821,6 +1821,160 @@ def apply_box(df_exp):
+                      dst_global_extent=None,
+                      dst_buffer=0, resampling=None, conserve=None,
+                      **kwargs):
+    """
+    Convenience wrapper function around rasterio.warp.reproject :
+    Reproject (up- or downsample) 2D np.ndarray to a grid using rasterio.
+    The destination (target) grid is defined from dst_transform and optionally
+    dst_resolution, dst_global_origin, and dst_crs.
+    This function ensures that reprojected data with the same dst_resolution and
+    dst_global_origins are on the same global grid, i.e., no offset between
+    destination grid points for different source grids that are projected to
+    the same target resolution.
+
+    Parameters
+    ----------
+    source : np.ndarray
+        2D-array containing the values to be reprojected
+    src_crs : CRS
+        Source CRS
+    src_transform : Affine
+        Source transform, contains info on grid, e.g.:
+            Affine(0.00833333333333333, 0.0, -18.175000000000068,
+                                 0.0, -0.00833333333333333, 43.79999999999993)
+    dst_crs : CRS, optional
+        Destination CRS. The default is src_crs.
+    dst_transform : Affine, optional
+        Destination transform. The default is src_transform.
+    dst_shape : tuple with integers, optional
+        (height, width) of destination grid. The default is source.shape.
+        Note: Destination shape is changed if resolution changes
+    dst_resolution : number, optional
+        Destination resolution.
+        Overwrites resolution provided in dst_transform.
+        Unit of resolution depends on CRS;
+        Typically but not necessarily, the unit is degree lat, lon.
+        The default is dst_transform[0].
+    dst_global_origin : tuple of numbers, optional
+        global lon and lat origins as basis for destination grid.
+        Only required if dst_resolution different than dst_transform[0].
+        The default is (-180.0, 90), change dst_global_origin for different units.
+    dst_global_extent : tuple of numbers, optional
+        Extent of global grid in unit of resoution.
+        The default is (180, 360). Change for different units than degree lat, lon.
+    dst_buffer : int, optional
+        number of grid cells added as buffer around data in target grid
+        (only applied if resolution changes). The default is 0.
+    resampling : resampling function or str, optional
+        The default is rasterio.warp.Resampling.nearest.
+        strings 'nearest', 'bilinear', and 'cubic' are excepted as well.
+    conserve : str, optional
+        either 'mean' or 'sum' or 'norm'
+        Conserve mean or sum of data or normalize?
+        The default is None (no conservation).
+
+    **kwargs : **kwargs
+        extra arguments for rasterio.warp.reproject.
+
+    Returns
+    -------
+    destination
+        numpy array.
+    meta_dst : dict
+        meta dict of destination.
```

Comment:
We only need to return the `dst_transform` because the `dst_crs` and `dst_shape` are known to the user. Furthermore, `destination` and `dst_transform` are also the return values of `rasterio.warp.reproject`.